File: website/Entities/VolunteerWithLogons.cs
using System;
using System.Collections.Generic;

namespace website.Entities
{
  public class VolunteerWithLogons
  {
    public string VolunteerId { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public List<string> Classes { get; set; } = new List<string>();
    public List<int> Grades { get; set; } = new List<int>();
    public List<Logon> Logons { get; set; } = new List<Logon>();
    public DateTime FirstLoginDate { get; set; }
    public DateTime LastLoginDate { get; set; }

    public class Logon 
    {
      public DateTime LogonTime { get; set; }
      public DayOfWeek DayOfWeek { get; set; }
    }
  }
}

File: HomeReadingLibraryWeb/ClientApp/src/app/entities/volunteer-with-logons.ts
import { VolunteerLogon } from './volunteer-logon';
export interface VolunteerWithLogons {
    voluteerId: string;
    firstName: string;
    lastName: string;
    classes: string[];
    grades: string[];
    logons: VolunteerLogon[];
    firstLoginDate: Date;
    lastLoginDate: Date;
}
